src/main/java/org/folio/dcb/service/impl/LendingLibraryServiceImpl.java

```diff
@@ -62,4 +65,35 @@ private void saveDcbTransaction(String dcbTransactionId, DcbTransaction dcbTrans
     transactionRepository.save(transactionEntity);
   }
 
+  @Override
+  public void updateTransactionStatus(String checkInEvent) {
+    var checkInItemId = parseCheckInEvent(checkInEvent);
+    log.info("updateTransactionStatus:: Received checkIn event for itemId: {}", checkInItemId);
+
+    if (Objects.nonNull(checkInItemId)) {
+      transactionRepository.findByItemId(checkInItemId)
+        .ifPresent(transactionEntity -> {
+          if (transactionEntity.getStatus().equals(TransactionStatus.StatusEnum.CREATED)) {
+            transactionEntity.setStatus(TransactionStatus.StatusEnum.OPEN);
```

Comment:
Do we need to check for Library role over here ? Or this transaction status will updated for all 3 libraries ? Please check status-transition diagram - https://app.diagrams.net/#G1Ps2_ohN0FfEM0PHenVLrEdCG0Sop2ZxT#%7B%22pageId%22%3A%22H0niUHbnuylXihCLpYzt%22%7D